src/bluetooth/DataAccess.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2021
+ */
+
+/**
+ * Helper methods to perform data access functionality for breach calculations
+ */
+import VACCINE_ENTITIES from '../utilities/modules/vaccines/constants';
+
+export class DataAccess {
+  constructor(dbService) {
+    this.db = dbService;
+  }
+
+  getTemperatureLogsFrom = (sensorId, timeToCheckFrom) =>
+    this.db
+      .objects(VACCINE_ENTITIES.TEMPERATURE_LOG)
+      .filtered('sensor.id == $0 AND timestamp >= $1', sensorId, timeToCheckFrom)
+      .sorted('timestamp');
+
+  getMostRecentBreachLog = sensorId => {
```

Comment:
This should be the most recent log which was a part of a breach. 

- really it's used to determine when to find logs 'from'. I.e. if I download a log now, but to trigger a breach we need 1 hour of logs, we need to chec previous logs to determine if 'this is the one' which will create a breach. 
- Technically, we can look backwards and check only the duration of a breach - i.e. if a breach requires 1 hour worth of logs, we would technically only need to look back 1 hour.
-  It's just easier (imo!) to always look back to the most recent breached log (where we can guarantee that either this new log that has just been downloaded is a part of the same breach, or all of the logs since that breach could make up the new one) rather than looking up all the breach configs, determining which is the highest duration and looking back that amount. It's certainly not the most efficient, though!


So, I believe adding `AND temperatureBreach != $2, ..., null)` (I think, I always forget how to check for not exists... maybe `NOT null`..?)

Reply:
Whoops I just realised this method is just grabbing _all_ the temperature logs and returning the most recent 🤦 

Using NOT seemed to throw an invalid predicate error, but != appears to be at least a valid query (is there an easier way to manually populate the DB with data or is spamming create record methods the way to go?)

4aaf415b

Reply:
I think @Chris-Petty was working on a way to inject records easily ;-)

But really - we should probably make a test db, but in this epic is a zip `createData.zip` which is some 4D code to create records, which you could sync to your store.

But seems to be how you do it! https://github.com/sussol/msupply/issues/7271#issue-734959779